### DISCUSSION TITLE
Separate the modelling from the likelihood functionality

### DISCUSSION THREAD

**Discussion by tilmantroester** [LSSTDESC/firecrown#347] [Ideas]
The way firecrown is structured currently strongly assumes that one wants to calculate a log likelihood. Often we only want to make some model predictions, which do not depend on the likelihood, data vectors, or covariances.

`compute_theory_vector` should only depend on things actually required to evaluate the theory vector. So in the case of two-point statistics, the n(z) and ell/theta ranges/binning, but not for example the covariance.

A step further would be to provide some utility to bootstrap a minimal sacc object from some more basic quantities, like type of statistic, a list of n(z), and ell bins, so that the user doesn't have to deal with sacc files at all if they just want a model prediction.

This is related to the discussion in #308 on standardising model specifications.

**Comment by marcpaterno** [comment]
We think what you have asked for can now be done with the current release (v1.12) of Firecrown.
Is there something else that is needed for this particular issue?
If not, we'd like to close this discussion.
